src/circuit.rs

```diff
@@ -2,55 +2,669 @@
+
+        // Configuration for curve point operations.
+        // This uses 10 advice columns and spans the whole circuit.
+        let ecc_config = EccChip::configure(meta, advices, table_idx, ecc_constants);
+
+        // Configuration for the Poseidon hash.
+        let poseidon_config = PoseidonChip::configure(
+            meta,
+            poseidon::OrchardNullifier,
+            [advices[0], advices[1], advices[2]],
+            advices[3],
+        );
+
+        // Configuration for standard PLONK (addition and multiplication).
+        let plonk_config = PLONKChip::configure(meta, [advices[0], advices[1], advices[2]]);
+
+        // Configuration for a Sinsemilla hash instantiation and a
+        // Merkle hash instantiation using this Sinsemilla instance.
+        // Since the Sinsemilla config uses only 5 advice columns,
+        // we can fit two instances side-by-side.
+        let (sinsemilla_config_1, merkle_config_1) = {
+            let sinsemilla_config_1 = SinsemillaChip::configure(
+                meta,
+                advices[..5].try_into().unwrap(),
+                lookup,
+                sinsemilla_1_constants,
+            );
+            let merkle_config_1 = MerkleChip::configure(meta, sinsemilla_config_1.clone());
+
+            (sinsemilla_config_1, merkle_config_1)
+        };
+
+        // Configuration for a Sinsemilla hash instantiation and a
+        // Merkle hash instantiation using this Sinsemilla instance.
+        // Since the Sinsemilla config uses only 5 advice columns,
+        // we can fit two instances side-by-side.
+        let (sinsemilla_config_2, merkle_config_2) = {
+            let sinsemilla_config_2 = SinsemillaChip::configure(
+                meta,
+                advices[5..].try_into().unwrap(),
+                lookup,
+                sinsemilla_2_constants,
+            );
+            let merkle_config_2 = MerkleChip::configure(meta, sinsemilla_config_2.clone());
+
+            (sinsemilla_config_2, merkle_config_2)
+        };
+
+        // Configuration to handle decomposition and canonicity checking
+        // for CommitIvk.
+        let commit_ivk_config =
+            CommitIvkConfig::configure(meta, advices, sinsemilla_config_1.clone());
+
+        // Configuration to handle decomposition and canonicity checking
+        // for NoteCommit_old.
+        let old_note_commit_config =
+            NoteCommitConfig::configure(meta, advices, sinsemilla_config_1.clone());
+
+        // Configuration to handle decomposition and canonicity checking
+        // for NoteCommit_new.
+        let new_note_commit_config =
+            NoteCommitConfig::configure(meta, advices, sinsemilla_config_2.clone());
+
+        // TODO: Infrastructure to handle public inputs.
+        let q_primary = meta.selector();
+        let primary = meta.instance_column();
 
         // Placeholder gate so there is something for the prover to operate on.
```

Comment:
Why do we still need this?

Reply:
@therealyingtong is still in the process of migrating this PR to use the new `halo2` public input APIs.